doc/dev/design/workflows/storage_devices_and_osds.rst

```diff
@@ -0,0 +1,304 @@
+==============================================
+Storage devices and OSD’s management workflows
+==============================================
+The cluster storage devices are the physical storage devices installed in each of the cluster’s hosts. We need to execute different operations over them and also to retrieve information about physical features and working behaviour.
+The basic use cases we have in this area are:
+
+- `1. Retrieve device information. Inventory`_
+- `2. Add OSDs`_
+- `3. Remove OSDs`_
+
+1. Retrieve device information. Inventory
+=========================================
+We must be able to review what is the current state and condition of the cluster storage devices. We need the identification and features detail(including ident/fault led on/off capable) and if the device is used or not as an OSD/DB/WAL device.
+
+The information required for each device will be at least:
+::
+
+    Hostname   Path      Type  Serial    Size   Health  Ident  Fault  Available
+
+.. Note: A more optional extended view with other information fields could be also useful.
+
+In order to know what is the current condition of the device, we need to know what is the amount of storage used, the percentage of free space, the average number of iops and  the fault led state. Part of the operational information for each device has its source in the data gathered by Prometheus. Given that the monitoring stack has reached a very important role in the Ceph Dashboard, it seems sensible to start to think in the monitoring stack as a fundamental piece of the UI installed by default.  Once we have Prometheus as an “always present” source of information, it seems sensible to start to use part of the information about performance and operational state in all the places where it can be useful to the final user.
```

Comment:
Note that to my knowledge it's not possible to *obtain* the state of the fault LED, it can only be set manually. I agree that a device's IOPs is a key metric and highly desirable, but can we actually determine these? I'm not aware that e.g. Prometheus is providing this information on a device level.

With regards to relying on Prometheus here: do you envision that Ceph Dashboard is pulling these metrics from Prometheus directly, in a similar fashion to how Grafana is doing it?

Reply:
> Note that to my knowledge it's not possible to _obtain_ the state of the fault LED, it can only be set manually. I agree that a device's IOPs is a key metric and highly desirable, but can we actually determine these? I'm not aware that e.g. Prometheus is providing this information on a device level.

I do not know, what i'm saying is "if this is a feature (to know directly the fault led state)  that will provide value ,.. lets go for it ... "

> With regards to relying on Prometheus here: do you envision that Ceph Dashboard is pulling these metrics from Prometheus directly, in a similar fashion to how Grafana is doing it?

yes. It is worth to have this information directly as the minimum set of devices attributes? Should we start to deem prometheus as part of the dashboard product?

Reply:
Removed "prometheus" from the design document. We should rely in the orchestrator to retrieve the operational data from the devices. (Probably using ceph volume, and other alternative  sources. )
What is clear, is that there is a consensus about not to be linked to "prometheus"  or other third party tools.